src/Microsoft.Azure.WebJobs.Host/Scale/ScaleOptions.cs

```diff
@@ -0,0 +1,106 @@
+            // Assume a case of 100 functions in an app, each mapping to a monitor. Thats
+            // 1200 samples to read from storage on each scale status request.
+            ScaleMetricsMaxAge = TimeSpan.FromMinutes(2);
+            ScaleMetricsSampleInterval = TimeSpan.FromSeconds(10);
+            MetricsPurgeEnabled = true;
+            CheckTargetScalerEnabled = ts => false;
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating the maximum age for metrics.
+        /// Metrics that exceed this age will not be returned to monitors.
+        /// </summary>
+        public TimeSpan ScaleMetricsMaxAge
+        {
+            get
+            {
+                return _scaleMetricsMaxAge;
+            }
+
+            set
+            {
+                if (value < TimeSpan.FromMinutes(1) || value > TimeSpan.FromMinutes(5))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(ScaleMetricsMaxAge));
+                }
+                _scaleMetricsMaxAge = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the sampling interval for metrics.
+        /// </summary>
+        public TimeSpan ScaleMetricsSampleInterval
+        {
+            get
+            {
+                return _scaleMetricsSampleInterval;
+            }
+
+            set
+            {
+                if (value < TimeSpan.FromSeconds(1) || value > TimeSpan.FromSeconds(30))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(ScaleMetricsSampleInterval));
+                }
+                _scaleMetricsSampleInterval = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether old metrics data
+        /// will be auto purged.
+        /// </summary>
+        public bool MetricsPurgeEnabled { get; set; }
+
+        public string Format()
+        {
+            var options = new JObject
+            {
+                { nameof(ScaleMetricsMaxAge), ScaleMetricsMaxAge },
+                { nameof(ScaleMetricsSampleInterval), ScaleMetricsSampleInterval },
+                { nameof(MetricsPurgeEnabled), MetricsPurgeEnabled },
+                { nameof(IsTargetScalingEnabled), IsTargetScalingEnabled }
+            };
+
+            return options.ToString(Formatting.Indented);
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether target base scaling is enabled at the host level.
+        /// </summary>
+        public bool IsTargetScalingEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets a function to check if a particular <see cref="ITargetScaler"/> is enabled.
+        /// </summary>
+        public Func<ITargetScaler, bool> CheckTargetScalerEnabled { get; set; }
+
+        /// <summary>
+        /// Gets or sets the function metadata to configure the host with.
+        /// </summary>
+        public JArray FunctionMetadata { get; set; }
```

Comment:
This is a temporary hack - we'll need to introduce an interface for providing function metadata to the host